### ISSUE TITLE
sst_big_num Constructor from String is Broken on POWER8

### ISSUE THREAD

**Issue by nmhamster** [sstsimulator/sst-core#294]
Do not reliability get back a multiplier by SI unit in UnitAlgebra. The issue appears to be the `sst_big_num` does not work when constructing from string.

```
UnitAlgebra: [1][ps]
PRE-COMPOND CHECK: units=ps, si_length=1
si_unit=p, doDouble=1.000000
No compound unit
ps

Units after find: [s]
Multipler=[1]
Value=[1]
UA Print=[1 s]
UnitAlgebra: [1][ns]
PRE-COMPOND CHECK: units=ns, si_length=1
si_unit=n, doDouble=1.000000
No compound unit
```

**Comment by bjmoor** [comment]
Appears to be due to the undefined cast from double ("-18") to unsigned int in `decimal_fixedpoint` returning 0.  Works on x86 due to the cast returning 0xffffffee, which later causes math to wrap-around.

@feldergast is planning on taking a look today.

**Comment by nmhamster** [comment]
Addressed in #298. Reopen is problem reoccurs.
